packages/hub/src/lib/snapshot-download.ts

```diff
@@ -0,0 +1,119 @@
+import type { CredentialsParams, RepoDesignation } from "../types/public";
+import { listFiles } from "./list-files";
+import { getHFHubCachePath, getRepoFolderName } from "./cache-management";
+import { spaceInfo } from "./space-info";
+import { datasetInfo } from "./dataset-info";
+import { modelInfo } from "./model-info";
+import { toRepoId } from "../utils/toRepoId";
+import { join, dirname } from "node:path";
+import { mkdir, writeFile } from "node:fs/promises";
+import { downloadFileToCacheDir } from "./download-file-to-cache-dir";
+
+export const DEFAULT_REVISION = "main";
+
+export async function snapshotDownload(
```

Comment:
Can you add a doc comment - saying it downloads a revision of a repo to local file system? And that it returns the local path. And that there are related methods (`downloadFileToCacheDir`, and `scanCachedRepo`) ?

(thinking about it, we could also do web storage with [indexed db](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Client-side_storage#storing_complex_data_%E2%80%94_indexeddb) - but let's do that if ever in another PR :sweat: )